### PR TITLE
fix(groups): Group membership request check existing requests

### PR DESCRIPTION
**Pull request by lowfill**
Added check of exisiting request and added error message when an existing request exists

Fixes: #8901

### CODE REVIEW

File: mod/groups/actions/groups/membership/join.php [hunk 1]
```diff
@@ -44,35 +44,40 @@
 			register_error(elgg_echo("groups:cantjoin"));
 		}
 	} else {
-		add_entity_relationship($user->guid, 'membership_request', $group->guid);
+		if(!check_entity_relationship($user->guid,'membership_request',$group->guid)){
```
Comment (hypeJunction) [inline mod/groups/actions/groups/membership/join.php]: We prefer early termination:
```
if () {

} else if (check_entity_relationship()) {
   register_error();
   forward(REFERRER);
} else {

}
```

File: mod/groups/languages/en.php [hunk 2]
```diff
@@ -134,6 +134,7 @@
 	'groups:joinrequestmade' => 'Requested to join group',
 	'groups:joined' => 'Successfully joined group!',
 	'groups:left' => 'Successfully left group',
+	'groups:requestexists' => 'You alredy requested join this group, please wait the group owner to respond.',
```
Comment (hypeJunction) [inline mod/groups/languages/en.php]: `You have already requested to join this group`, should be enough.

File: mod/groups/actions/groups/membership/join.php [hunk 3]
```diff
@@ -44,35 +44,40 @@
 			register_error(elgg_echo("groups:cantjoin"));
 		}
 	} else {
-		add_entity_relationship($user->guid, 'membership_request', $group->guid);
```
Comment (mrclay) [inline mod/groups/actions/groups/membership/join.php]: Instead use a return early strategy, which (in addition to other benefits) makes the diff cleaner to read. E.g.

```php
if (check_entity_relationship($user->guid, 'membership_request', $group->guid)) {
	register_error(elgg_echo("groups:requestexists"));
	forward(REFERER);
}
```

Also we enforce a [code style](http://learn.elgg.org/en/stable/contribute/code.html#php-guidelines) close to PSR-2. The Travis build will tell you if you get it wrong.

File: mod/groups/languages/en.php (hunk 2 above)
Comment (mrclay) [inline mod/groups/languages/en.php]: `You already requested to join this group. Please wait for the group owner to respond.`

**Review by mrclay — CHANGES_REQUESTED** [review]

**Comment by mrclay** [comment]
Oops, 2 reviews saying same thing :)

File: mod/groups/languages/en.php (hunk 2 above)
Comment (mrclay) [inline mod/groups/languages/en.php]: Agreed.

File: mod/groups/languages/en.php [hunk 4]
```diff
@@ -134,6 +134,7 @@
 	'groups:joinrequestmade' => 'Requested to join group',
 	'groups:joined' => 'Successfully joined group!',
 	'groups:left' => 'Successfully left group',
+	'groups:requestexists' => 'You alredy requested join this group.',
```
Comment (mrclay) [inline mod/groups/languages/en.php]: `You already requested to join this group.`

File: mod/groups/actions/groups/membership/join.php [hunk 5]
```diff
@@ -43,7 +43,11 @@
 		} else {
 			register_error(elgg_echo("groups:cantjoin"));
 		}
-	} else {
+	} else if(check_entity_relationship($user->guid,'membership_request',$group->guid)) {
```
Comment (mrclay) [inline mod/groups/actions/groups/membership/join.php]: - space after `if` (not a function call)
- space after each `,` in argument list
- `} else {`, not Drupal style

**Comment by lowfill** [comment]
I don't understand was Travis is saying

**Comment by mrclay** [comment]
See https://gitter.im/Elgg/Elgg?at=57e67afe00df3cc4224173dc

File: mod/groups/languages/en.php (hunk 2 above)
Comment (jdalsem) [inline mod/groups/languages/en.php]: typo in alredy..

File: mod/groups/languages/en.php [hunk 6]
```diff
@@ -134,6 +134,7 @@
 	'groups:joinrequestmade' => 'Requested to join group',
 	'groups:joined' => 'Successfully joined group!',
 	'groups:left' => 'Successfully left group',
+	'groups:requestexists' => 'You already requested join this group.',
```
Comment (mrclay) [inline mod/groups/languages/en.php]: "requested **to** join"

Comment (hypeJunction) [inline mod/groups/languages/en.php]: I think proper English is `You have already requested` if no time is specified. Past tense is used if you specify time `You requested to joint his group 2 days ago`.

**Comment by hypeJunction** [comment]
@lowfill Thanks for your contribution. I have used it as a basis for a different pull request here: #10379 10379

**Pull request closed without merging**